eval_model.py

```diff
@@ -1,27 +1,66 @@
-from architecture import Model1
+from architecture import Model1, load_and_modify_resnet18, LeNet
 from util import log, Config
 import torch
 import os
+import numpy
 from data_loader import make_evaluation_loader
+import matplotlib.pyplot as plt
+from sklearn.metrics import confusion_matrix, precision_score, recall_score, accuracy_score, f1_score
 
 
 def eval_model(model, testset_path):
-    test_loader = make_evaluation_loader(testset_path)
 
-    def get_class(idx):
-        return test_loader.dataset.classes[idx]
+    def plot_confusion_matrix(conf_matrix, title):
+        plt.imshow(conf_matrix, interpolation='nearest', cmap=plt.cm.Blues)
+        plt.title(title)
+        plt.colorbar()
+        classes = ['Masked', 'Unmasked']
```

Comment:
This does not need to be hardcoded imo, pretty sure you can extract the classes from `test_loader.dataset.classes`. It should be a dict IIRC, then you can just call `values()` or whatever the python equivalent is.

Reply:
Yeah I thought about it - it's just a hackie solution and If you don't mind I would keep it like that because right now I don't have a trained model to test any changes on. Otherwise I can change it tomorrow. 

We also could mention it in the reflection section of the report 😅

Reply:
Should not be a big change, all you gotta do is move the construction of the test loader to above the function definition so the closure can reference it and then make the following change

```suggestion
        classes = test_loader.dataset.classes
```

IDK, maybe I am missing the point/issue, so I will leave it at that.

Reply:
I just didn't want to change it since I have no way of testing it rn (I don't have a model to evaluate at the moment).
 
Your solution looks fine to me tho and I will check it once a training is completed.

Reply:
We already have a demo model ready for download (just run `make model`) :)

Either way, this is so inconsequential and can easily be changed later. I will go ahead and approve now.